spccore/internal/lock.py

```diff
@@ -176,6 +176,29 @@ def _update_lock_time(self):
 
     # Make the lock object a Context Manager
     def __enter__(self):
+        """
+        Important notes:
+
+        We implemented __enter__ to support using lock with the context manager syntax:
+
+        with Lock("foo"):
+            <do something while holding the lock>
+
+        However, using nested context manager for the same lock will cause the inner context manager to release the lock
```

Comment:
However, using nested context manager for the same lock will cause the inner context manager to wait for the out lock to be released before it can acquire the lock.